core/src/main/java/org/apache/iceberg/UpdateRequirements.java

```diff
@@ -117,6 +141,11 @@ private void update(MetadataUpdate.AddSchema update) {
       if (!addedSchema) {
         if (base != null) {
           require(new UpdateRequirement.AssertLastAssignedFieldId(base.lastColumnId()));
+        } else if (viewMetadataBase != null) {
+          // TODO: is this correct here? we don't track lastColumnId in ViewMetadata
```

Comment:
See the comment above. Short answer is that I don't think we actually need _any_ assertions right now.